tool/client.cc

```diff
@@ -540,24 +682,58 @@ bool Client(const std::vector<std::string> &args) {
     SSL_CTX_set_permute_extensions(ctx.get(), 1);
   }
 
+  std::string certPathFlag;
+  int verify = SSL_VERIFY_NONE;
   if (args_map.count("-root-certs") != 0) {
+    certPathFlag = "-root-certs";
+    verify = SSL_VERIFY_PEER;
+  }
+  // For the OpenSSL tool, simply specifying -CAfile does not imply verification
+  if (args_map.count("-CAfile") != 0) {
+    certPathFlag = "-CAfile";
+  }
+  if (!certPathFlag.empty()) {
     if (!SSL_CTX_load_verify_locations(
-            ctx.get(), args_map["-root-certs"].c_str(), nullptr)) {
+            ctx.get(), args_map[certPathFlag].c_str(), nullptr)) {
       fprintf(stderr, "Failed to load root certificates.\n");
       ERR_print_errors_fp(stderr);
       return false;
     }
-    SSL_CTX_set_verify(ctx.get(), SSL_VERIFY_PEER, nullptr);
   }
 
+  certPathFlag = "";
   if (args_map.count("-root-cert-dir") != 0) {
+    certPathFlag = "-root-cert-dir";
+    verify = SSL_VERIFY_PEER;
+  }
+  // For the OpenSSL tool, simply specifying -CApath does not imply verification
+  if (args_map.count("-CApath") != 0) {
```

Comment:
`-CApath` and `-CAfile` are not listed in `kArguments` at the top of the file.  Is that intentional?

Reply:
Yes this is intentional. We are utilizing the same body of logic for both the bssl and openssl tool. CApath and CAfile are for the openssl tool, root-cert and root-cert-dir are for the bssl tool (which are specified in this file).